### ISSUE TITLE
How to use this tool with NX?

### ISSUE THREAD

**Issue by NicolaLovo** [fi3ework/vite-plugin-checker#774]
### Describe the bug

I want to use this tool in my NX repository, but there is no documentation on that and all the attempts I made on the official NX templates do not work.

### Reproduction

1. Clone NX official template: https://github.com/nrwl/react-template
2. Install `npm i vite-plugin-checker -D`
3. Go to `vite.config.mts` of the shop frontend app and insert in the plugin array:

```
checker({
      typescript: true,
})
```

4. Now try to set any typescript error in any tsx file, like: `const a: string = 26;`
5. Build: `nx build shop`.
6. No error is flagged

Alternatively, if you configure checker as:
```
checker({
      typescript: {
        root: `${process.cwd()}/apps/shop`,
        tsconfigPath: "tsconfig.app.json",
}
})
```
or:
```
checker({
      typescript: {
        root: `${process.cwd()}`,
        tsconfigPath: "tsconfig.app.json",
}
})
```

It throws lots of errors like:

```
Output file '.../dist/index.d.ts' has not been built from source file '.../src/index.ts'.
```
Where the repository imports components defined in packages of the packages/ directory.

### Expected behavior

Checker should be working in NX

### System Info

```shell
System:
    OS: Windows 11 10.0.26200
    CPU: (6) x64 Intel(R) Core(TM) i5-9600K CPU @ 3.70GHz
    Memory: 12.82 GB / 31.93 GB
  Binaries:
    Node: 22.18.0 - C:\Program Files\nodejs\node.EXE
    npm: 10.9.3 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 149.0.7827.201
    Edge: Chromium (150.0.4078.48)
    Internet Explorer: 11.0.26100.8115


node -v
v22.18.0

npm -v
10.9.3
```

### Validations

- [x] Read the [docs](https://github.com/fi3ework/vite-plugin-checker#readme).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
